### ISSUE TITLE
Build fails on musl host

### ISSUE THREAD

**Issue by adrianisuru** [thecoshman/http#107]
When compiling https on `x86_64-unknown-linux-musl` build fails due a difference in the function signature of `ioctl`

```
$ rustc --version
rustc 1.43.1 (8d69840ab 2020-05-04)
```

```
$ cargo build
Compiling https v1.9.2 (/home/adrianisuru/http)
error[E0308]: mismatched types
  --> src/util/os/non_windows_non_macos.rs:39:47
   |
39 |             let ok = unsafe { ioctl(dev_file, BLKGETSIZE, &mut block_count as *mut c_ulong) } == 0;
   |                                               ^^^^^^^^^^ expected `i32`, found `u64`
   |
help: you can convert an `u64` to `i32` and panic if the converted value wouldn't fit
   |
39 |             let ok = unsafe { ioctl(dev_file, BLKGETSIZE.try_into().unwrap(), &mut block_count as *mut c_ulong) } == 0;
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> src/util/os/non_windows_non_macos.rs:39:47
   |
39 |             let ok = unsafe { ioctl(dev_file, BLKGETSIZE, &mut block_count as *mut c_ulong) } == 0;
   |                                               ^^^^^^^^^^ expected `i32`, found `u64`
   |
help: you can convert an `u64` to `i32` and panic if the converted value wouldn't fit
   |
39 |             let ok = unsafe { ioctl(dev_file, BLKGETSIZE.try_into().unwrap(), &mut block_count as *mut c_ulong) } == 0;
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
For more information about this error, try `rustc --explain E0308`.
error: could not compile `https`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: could not compile `https`.
```
On `glibc` systems the `ioctl` function signature is 
```c
extern int ioctl (int __fd, unsigned long int __request, ...) __THROW;
```
but on `musl` systems the signature is
```c
int ioctl (int, int, ...);
```
so compiling [`src/util/os/non_windows_non_macos.rs`](https://github.com/thecoshman/http/blob/master/src/util/os/non_windows_non_macos.rs) fails because `BLKGETSIZE` is a `c_ulong` not a `c_int`

**Comment by adrianisuru** [comment]
Update `http` and `httplz` segfault when they get a request. The program opens fine, but when I connect from my browser it segfaults.

**Comment by nabijaczleweli** [comment]
Closing since the issue was resolved via #108.

**Comment by nabijaczleweli** [comment]
Released in [`v1.9.3`](https://github.com/thecoshman/http/releases/tag/v1.9.3).
